--- libxml2-2.9.0/Makefile.in	2012-09-11 08:55:37.000000000 +0200
+++ libxml2-2.9.0-android/Makefile.in	2013-02-25 15:27:25.000000000 +0100
@@ -41,9 +41,9 @@ noinst_PROGRAMS = testSchemas$(EXEEXT) t
 	testSAX$(EXEEXT) testHTML$(EXEEXT) testXPath$(EXEEXT) \
 	testURI$(EXEEXT) testThreads$(EXEEXT) testC14N$(EXEEXT) \
 	testAutomata$(EXEEXT) testRegexp$(EXEEXT) testReader$(EXEEXT) \
-	testapi$(EXEEXT) testModule$(EXEEXT) runtest$(EXEEXT) \
+	testapi$(EXEEXT) testModule$(EXEEXT) \
 	runsuite$(EXEEXT) testchar$(EXEEXT) testdict$(EXEEXT) \
-	runxmlconf$(EXEEXT) testrecurse$(EXEEXT) testlimits$(EXEEXT)
+	runxmlconf$(EXEEXT) testlimits$(EXEEXT)
 bin_PROGRAMS = xmllint$(EXEEXT) xmlcatalog$(EXEEXT)
 subdir = .
 DIST_COMMON = README $(am__configure_deps) $(srcdir)/Makefile.am \
