--- gnustep-base-1.24.0/Source/GSLocale.m	2011-09-29 21:00:46.000000000 +0200
+++ gnustep-base-1.24.0-android/Source/GSLocale.m	2013-02-27 15:09:16.000000000 +0100
@@ -161,6 +161,11 @@ GSDomainFromDefaultLocale(void)
 	   forKey: NSTimeFormatString];
 #endif /* HAVE_LANGINFO_H */
 
+  /**
+   * Android define an empty struct lconv and
+   * localeconv is just define in locale.h but don't exist in libc
+   */
+#ifndef __ANDROID__
   lconv = localeconv();
 
   /* Currency Information */
@@ -195,6 +200,7 @@ GSDomainFromDefaultLocale(void)
       [dict setObject: ToString(lconv->thousands_sep)
 	       forKey: NSThousandsSeparator];
     }
+#endif /* __ANDROID__ */
 
   /* FIXME: Get currency format from localeconv */
 
