--- gnustep-base-1.24.0/Source/NSPathUtilities.m	2011-11-14 11:01:24.000000000 +0100
+++ gnustep-base-1.24.0-android/Source/NSPathUtilities.m	2013-03-09 20:33:02.000000000 +0100
@@ -1803,6 +1803,7 @@ NSFullUserName(void)
 #else
 #ifdef  HAVE_PWD_H
 #if     defined(HAVE_GETPWNAM_R)
+#if     defined(HAVE_PW_GECOS_IN_PASSWD)
       struct passwd pw;
       struct passwd *p;
       char buf[BUFSIZ*10];
@@ -1814,8 +1815,10 @@ NSFullUserName(void)
               userName = [NSString stringWithUTF8String: pw.pw_gecos];
 	    }
         }
+#endif /* HAVE_PW_GECOS_IN_PASSWD */
 #else
 #if     defined(HAVE_GETPWNAM)
+#if     defined(HAVE_PW_GECOS_IN_PASSWD)
       struct passwd	*pw;
 
       [gnustep_global_lock lock];
@@ -1825,6 +1828,7 @@ NSFullUserName(void)
           userName = [NSString stringWithUTF8String: pw->pw_gecos];
         }
       [gnustep_global_lock lock];
+#endif /* HAVE_PW_GECOS_IN_PASSWD */
 #endif /* HAVE_GETPWNAM */
 #endif /* HAVE_GETPWNAM_R */
 #endif /* HAVE_PWD_H */
