--- gnustep-base-1.24.0/Source/NSDate.m	2011-11-04 11:38:16.000000000 +0100
+++ gnustep-base-1.24.0-android/Source/NSDate.m	2013-02-27 14:46:05.000000000 +0100
@@ -1233,7 +1233,15 @@ otherTime(NSDate* other)
 - (NSTimeInterval) timeIntervalSinceDate: (NSDate*)otherDate
 {
   if (otherDate == nil)
+#ifndef __ANDROID__
     return nan("");
+#else
+    /* Android don't have nan, nanl or nanf */
+    {
+        [NSException raise: NSInvalidArgumentException
+                    format: @"nil argument for timeIntervalSinceDate:"];
+    }
+#endif
 /*
     {
       [NSException raise: NSInvalidArgumentException
