--- gnustep-base-1.24.0/Source/GSFileHandle.m	2011-12-15 10:42:39.000000000 +0100
+++ gnustep-base-1.24.0-android/Source/GSFileHandle.m	2013-02-23 11:00:02.000000000 +0100
@@ -75,7 +75,7 @@
 /*
  *	Stuff for setting the sockets into non-blocking mode.
  */
-#ifdef	__POSIX_SOURCE
+#if defined(__POSIX_SOURCE) || defined(__ANDROID__)
 #define NBLK_OPT     O_NONBLOCK
 #else
 #define NBLK_OPT     FNDELAY
--- gnustep-base-1.24.0/Source/NSMessagePort.m	2012-01-09 09:28:27.000000000 +0100
+++ gnustep-base-1.24.0-android/Source/NSMessagePort.m	2013-02-23 11:01:23.000000000 +0100
@@ -77,7 +77,7 @@
 /*
  *	Stuff for setting the sockets into non-blocking mode.
  */
-#ifdef	__POSIX_SOURCE
+#if defined(__POSIX_SOURCE) || defined(__ANDROID__)
 #define NBLK_OPT     O_NONBLOCK
 #else
 #define NBLK_OPT     FNDELAY
--- gnustep-base-1.24.0/Source/NSSocketPort.m	2012-01-09 09:28:27.000000000 +0100
+++ gnustep-base-1.24.0-android/Source/NSSocketPort.m	2013-02-23 10:56:07.000000000 +0100
@@ -84,7 +84,7 @@
 /*
  *	Stuff for setting the sockets into non-blocking mode.
  */
-#ifdef	__POSIX_SOURCE
+#if defined(__POSIX_SOURCE) || defined(__ANDROID__)
 #define NBLK_OPT     O_NONBLOCK
 #else
 #define NBLK_OPT     FNDELAY
--- gnustep-base-1.24.0/Source/NSThread.m	2011-12-15 10:42:39.000000000 +0100
+++ gnustep-base-1.24.0-android/Source/NSThread.m	2013-02-23 10:57:29.000000000 +0100
@@ -56,7 +56,7 @@
 #  include <fcntl.h>
 #endif
 
-#ifdef	__POSIX_SOURCE
+#if defined(__POSIX_SOURCE) || defined(__ANDROID__)
 #define NBLK_OPT     O_NONBLOCK
 #else
 #define NBLK_OPT     FNDELAY
