diff --git a/Source/GNUmakefile.in b/Source/GNUmakefile.in
index d410253cb11d7ab777d17603bce6be3f2f026bac..0454992e58c69013aab3a398a2f656c89d1d9e44 100644
--- a/Source/GNUmakefile.in
+++ b/Source/GNUmakefile.in
@@ -100,6 +100,7 @@ libgnustep-corebase_OBJC_FILES = \
   NSCFType.m
 
 libgnustep-corebase_LIBRARIES_DEPEND_UPON += @LIBS@
+libgnustep-corebase_LIBRARIES_DEPEND_UPON += -lgnustl_shared
 libgnustep-corebase_HEADER_FILES_DIR = ../Headers/CoreFoundation
 libgnustep-corebase_HEADER_FILES_INSTALL_DIR = CoreFoundation
 
