diff --git a/makeconf/makeconf/androidproject.rb b/makeconf/makeconf/androidproject.rb
index dfc737ceefc7894eba19441049e9a516160c8171..7e47882f03d7e46b2c47cdff1efb6d5be0194fc2 100644
--- a/makeconf/makeconf/androidproject.rb
+++ b/makeconf/makeconf/androidproject.rb
@@ -331,7 +331,7 @@ private
     if RbConfig::CONFIG['host_os'] =~ /linux/
       build_os = 'linux-x86';
     elsif RbConfig::CONFIG['host_os'] =~ /darwin/
-      build_os = 'darwin-x86';
+      build_os = 'darwin-x86_64';
     else
       throw 'Unknown build OS: ' + RbConfig::CONFIG['host_os']
     end
diff --git a/makeconf/makeconf/platform.rb b/makeconf/makeconf/platform.rb
index a1742743a9dda42b479af43a924f30a2aecf64df..52bedd95b1698c85f2490e566cf025eb0beeec6e 100644
--- a/makeconf/makeconf/platform.rb
+++ b/makeconf/makeconf/platform.rb
@@ -29,7 +29,8 @@ class Platform
 
   # Returns true or false depending on if the target is Linux
   def Platform.is_linux?
-    @@target_os =~ /^linux/
+    true
+    #@@target_os =~ /^linux/
   end
 
   # Returns true or false depending on if the target is x86-compatible
