diff --git CMakeLists.txt CMakeLists.txt
index eeaa4ae..043e0d5 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -210,7 +210,6 @@ if (ENABLE_OBJCXX)
 			add_library(objcxx SHARED ${libobjcxx_CXX_SRCS})
 			set_target_properties(objcxx PROPERTIES
 				LINKER_LANGUAGE C
-				SOVERSION ${libobjc_VERSION}
 				)
 			target_link_libraries(objcxx ${CXX_RUNTIME})
 			set(CXX_RUNTIME "")
@@ -226,7 +225,6 @@ if (ENABLE_OBJCXX)
 		add_library(objcxx SHARED ${libobjcxx_CXX_SRCS})
 		set_target_properties(objcxx PROPERTIES
 			LINKER_LANGUAGE C
-			SOVERSION ${libobjc_VERSION}
 			)
 		set(CXX_RUNTIME "")
 		list(APPEND INSTALL_TARGETS objcxx)
@@ -247,7 +245,6 @@ add_library(objc SHARED ${libobjc_C_SRCS} ${libobjc_ASM_SRCS} ${libobjc_OBJC_SRC
 
 set_target_properties(objc PROPERTIES
 	LINKER_LANGUAGE C
-	SOVERSION ${libobjc_VERSION}
 	OUTPUT_NAME ${LIBOBJC_NAME}
 	LINK_FLAGS "${objc_LINK_FLAGS}"
 	)
