Index: Test/BoxedForeignException.m
===================================================================
--- Test/BoxedForeignException.m	(revision 37420)
+++ Test/BoxedForeignException.m	(working copy)
@@ -1,8 +1,8 @@
 #define _GNU_SOURCE
+#include <stdlib.h>
 #include "../unwind.h"
 #include "Test.h"
 #include "../objc/hooks.h"
-#include <stdlib.h>
 
 
 struct foreign_exception
Index: Test/CMakeLists.txt
===================================================================
--- Test/CMakeLists.txt	(revision 37420)
+++ Test/CMakeLists.txt	(working copy)
@@ -12,7 +12,7 @@
 	BoxedForeignException.m
 	ExceptionTest.m
 	ForeignException.m
-	Forward.m
+	#Forward.m // test using TLS, not supported in android
 	ManyManySelectors.m
 	NestedExceptions.m
 	PropertyAttributeTest.m
@@ -21,7 +21,7 @@
 	ProtocolCreation.m
 	RuntimeTest.m
 	objc_msgSend.m
-	msgInterpose.m
+	#msgInterpose.m // test using TLS, not supported in android
 )
 
 # Function for adding a test.  This takes the name of the test and the list of
@@ -51,6 +51,6 @@
 endforeach()
 
 # Tests that are more than a single file.
-addtest_flags(CXXExceptions "-O0" "CXXException.m;CXXException.cc")
-addtest_flags(CXXExceptions_optimised "-O3" "CXXException.m;CXXException.cc")
+#addtest_flags(CXXExceptions "-O0" "CXXException.m;CXXException.cc")
+#addtest_flags(CXXExceptions_optimised "-O3" "CXXException.m;CXXException.cc")
 
